### ISSUE TITLE
twisted.protocols.amp._objectsToStrings makes a copy of a the `objects' dictionary in the least efficient way possible

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2665]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2665|
|Type|defect|
|Created|2007-05-19 23:24:10Z|

Attachments:

* [faster-objects-to-strings.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/470/47077c7cdd6c7822a07096245e1a47034c683618/65f3e06506b25064d89c0998dce1b40f757512e7.patch) (572 bytes) - added by exarkun on 2011-02-17 01:38:05Z - Copy the dict idiomatically and efficiently

<details><summary>Searchable metadata</summary>

```
trac-id__2665 2665
type__defect defect
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1179617050000000 1179617050000000
changetime__1297916127000000 1297916127000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Attached patch makes the copying less wasteful.  Running the AMP benchmark on my desktop, this produces a 20% speedup: 1768 req/sec against trunk; 2127 req/sec with the patch applied.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @exarkun|
|-|-|

Wow.  Incredibly simple and straightforward.  Passes all tests, deletes code, optimizes things.  I wonder what substance I was under the influence of which caused me to write it differently in the first place.  Please land post-haste!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [30789]) Speed up one corner of AMP argument serialization by simplifying the serialization code.

Author: exarkun
Reviewer: glyph
Fixes: [#2665](https://github.com/twisted/twisted/issues/2665)

Change some code in the implementation of AMP to just use `dict.copy` instead of a
slow reimplementation of that functionality.
